pkg/clients/cloudwatch/client.go

```diff
@@ -125,14 +123,12 @@ func (c Client) GetMetricStatistics(ctx context.Context, filter *cloudwatch.GetM
 	return resp.Datapoints
 }
 
-var _ CloudWatchClient = (*LimitedConcurrencyClient)(nil)
```

Comment:
Don't you think this might still be useful to double-check that the interface is implemented correctly?

Reply:
This is still being done just in a different way, `NewClient` and `NewLimitedConcurrencyClient` return the `Client` interface instead of their concrete types. The underlying `client` and `limitedConcurrencyClient` implementations are now private. This forces a consumer to use the new functions to obtain a client and ensures the clients implement the interface.